components/sup/src/manager/file_watcher.rs

```diff
@@ -1249,2216 +318,669 @@ impl<C: Callbacks, W: Watcher> FileWatcher<C, W> {
-    }
-
-    // Commands executed as a part of the test case step.
-    // Tests may come and go, so some of the variants may be unused.
-    #[allow(dead_code)]
-    #[derive(Debug)]
-    enum StepAction {
-        LnS(PathBuf, PathBuf),
-        MkdirP(PathBuf),
-        Touch(PathBuf),
-        Mv(PathBuf, PathBuf),
-        RmRF(PathBuf),
-        Nop,
-    }
-
-    #[derive(Clone, Copy, Debug, PartialEq)]
-    enum PathKind {
-        Symlink,
-        Regular,
-        MissingRegular,
-        Directory,
-        MissingDirectory,
     }
 
-    // Simplified description of the WatchedFile's Common struct.
-    #[derive(Clone)]
-    struct PathState {
-        kind:      PathKind,
-        path_rest: Vec<OsString>,
-        prev:      Option<PathBuf>,
-        next:      Option<PathBuf>,
-    }
-
-    impl IndentedToString for PathState {
-        fn indented_to_string(&self, spaces: &str, repeat: usize) -> String {
-            let mut formatter = IndentedStructFormatter::new("PathState", spaces, repeat);
-            formatter.add_debug("kind", &self.kind);
-            formatter.add_debug("path_rest", &self.path_rest);
-            formatter.add("prev", &self.prev);
-            formatter.add("next", &self.next);
-            formatter.fmt()
+    fn handle_file_disappearance(&mut self, notify_event: &Event) {
+        trace!("FileWatcher::disappearance");
+        for path_buf in notify_event.paths.iter() {
+            if self.file.contains(&path_buf) {
+                self.callbacks.file_disappeared(&path_buf)
+            } else {
+                self.log_ignored_path(notify_event, &path_buf);
+            }
         }
     }
 
-    #[derive(Clone, Copy, Debug, PartialEq)]
-    enum NotifyEventKind {
-        Appeared,
-        Modified,
-        Disappeared,
-    }
+    // Logging that a particular event was ignored.  Ignoring an event may be
+    // the correct thing to do so the purpose of logging that a particular event
+    // was ignored event is to highlight the fact that an event was ignored in
+    // case the decision made today to ignore that event needs to be changed 
+    // due to design changes, lessons learned (bugs) or notify crate updates.
+    fn log_ignored_event(&mut self, notify_event: &Event) { info!("IGNORING {:?}", notify_event) }
 
-    #[derive(Clone, Debug, PartialEq)]
-    struct NotifyEvent {
-        path: PathBuf,
-        kind: NotifyEventKind,
+    // Logging the fact that a particular path was ignored.  Ignoring a given
+    // path may be the correct thing to do so the purpose of logging that a
+    // particular path was ignored is to highlight that the path was ignored in
+    // case the decision made today to ignore the event needs to be changed 
+    // due to design changes, lessons learned (bugs) or notify crate updates.
+    fn log_ignored_path(&mut self, notify_event: &Event, path_buf: &PathBuf) {
+        info!("IGNORING {:?} of {:?}", path_buf, notify_event)
```

Comment:
this is probably something you want to log at TRACE level. INFO would be way too noisy.

Reply:
Will use a similar item below to track a final review of the logging levels.